rust/agama-lib/src/users/store.rs

```diff
@@ -43,21 +44,32 @@ impl UsersStore {
     pub async fn load(&self) -> Result<UserSettings, ServiceError> {
         let first_user = self.users_client.first_user().await?;
         let first_user = FirstUserSettings {
-            user_name: Some(first_user.user_name),
-            full_name: Some(first_user.full_name),
-            password: Some(first_user.password),
+            user_name: optional_string(first_user.user_name),
+            full_name: optional_string(first_user.full_name),
+            password: optional_string(first_user.password),
+            // sadly for boolean we do not have on dbus way to set it to not_set
             hashed_password: Some(first_user.hashed_password),
         };
+        let first_user = if first_user.skip_export() {
```

Comment:
Perhaps we could extend the D-Bus API with information about whether the user or the root was set instead of checking many conditions here.